src/_pytest/assertion/util.py

```diff
@@ -120,6 +120,12 @@ def isdict(x):
     def isset(x):
         return isinstance(x, (set, frozenset))
 
+    def isdatacls(obj):
+        return hasattr(obj, "__dataclass_fields__")
```

Comment:
please use `getattr(x, ..., None) is not None` or something like that instead, `hasattr` hides errors